project/SparkBuild.scala

```diff
@@ -765,16 +750,6 @@ object CopyDependencies {
 
 }
 
-object Java8TestSettings {
-  import BuildCommons._
-
-  lazy val settings = Seq(
-    javacJVMVersion := "1.8",
-    // Targeting Java 8 bytecode is only supported in Scala 2.11.4 and higher:
-    scalacJVMVersion := (if (System.getProperty("scala-2.10") == "true") "1.7" else "1.8")
```

Comment:
@vanzin maybe you know what I'm missing here, but the build seems to work fine with 2.10 without setting the `scalac` JVM version to 1.8. I used the `change-version-to-2.10.sh` script and build with `-Dscala-2.10=true` and saw that it was building `_2.10` artifacts with the Scala 2.10 libraries. I only have Scala 2.12 installed locally though, so I presume I'm cross-compiling. If the Scala code compiles vs 2.10 libraries, and is compiled by a `scalac` that successfully emits 1.8 bytecode, that should be OK? I feel like I'm missing something.